app/helpers/revisions_helper.rb

```diff
@@ -46,4 +46,67 @@ def link_to_conflicting_revision(record, revision)
     link_to text, record_revision_path(record, revision)
   end
 
+  def render_revision_object_icon(revision_object,object_type=nil)
+    object_type = revision_object_type(revision,revision_object) if object_type.nil?
+    # Set icon css depending on object type.
+    icon_css = %w{fa}
+    icon_css << case object_type
+                when 'Evidence'
+                  'fa-flag'
+                when 'Issue'
+                  'fa-bug'
+                when 'Note'
+                  'fa-file-text-o'
+                else
+                  ''
+                end
+    content_tag(:i, '', class: icon_css)
+  end
+
+  def render_revision_object_title(revision_object)
+    # Get title or content first characters.
+    if revision_object.fields.empty?
+      object_title = revision_object.respond_to?(:content) ? revision_object.content : revision_object.text
+    else
+      # Get title field, and if it's not set get first field value.
+      object_title = revision_object.title? ? revision_object.title : revision_object.fields.values[0]
+    end
+    content_tag(:span, truncate(object_title, length: 25, separator: "..."), class: 'item-content')
+  end
+
+  def revision_object_type(revision,revision_object=nil)
+    # If revision type is Note, check note's node id to determine object type.
+    if revision.item_type == 'Note'
+      revision_object = revision.reify if revision_object.nil?
+      object_type = revision_object.node_id == Node.issue_library.id ? 'Issue' : 'Note'
+    else
+      object_type = revision.item_type
+    end
+  end
+
+  def revision_object_location(revision_object,object_type)
+    object_type = revision_object_type(revision,revision_object) if object_type.nil?
+    # Get node if object is a Note or an Evidence.
+    if ['Note','Evidence'].include?(object_type)
+      if revision_object.node
+        "at " + link_to(revision_object.node.label,revision_object.node)
+      else
+        'at a Node which has since been deleted'
```

Comment:
I like the last part, but we should try to be consistent re `which has since been deleted`:

https://github.com/dradis/dradis-ce/blob/master/app/views/activities/_node.html.erb#L7

Unknown -> is strange (how come the app doesn't know?), and not 100% accurate.